lib/supavisor/health.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Supavisor.Health do
+  @moduledoc """
+  Health checking functionality
+
+  Current checks:
+  - Acceptable ERPC latencies: fails if a node has high latency to all
+  other nodes through :erpc.
+  - Database reachable: fails if can't run a simple query in the database.
+  """
+
+  require Logger
+
+  @checks [
+    acceptable_erpc_latencies: {__MODULE__, :acceptable_erpc_latencies?, []},
+    database_reachable: {__MODULE__, :database_reachable?, []}
+  ]
+
+  @task_supervisor __MODULE__.TaskSupervisor
+
+  @doc """
+  The main API for checking the health of a Supavisor node.
+  """
+  @spec health_check(Keyword.t()) :: :ok | {:error, :failed_checks, [atom()]}
+  def health_check(checks \\ @checks) do
+    successful_checks =
+      @task_supervisor
+      |> Task.Supervisor.async_stream_nolink(checks, fn {check, {m, f, a}} ->
+        result = apply(m, f, a)
+        {check, result}
+      end)
+      |> Enum.reduce([], fn result, acc ->
+        case result do
+          {:ok, {check, true}} -> [check | acc]
+          {:ok, {_check, false}} -> acc
+          {:exit, _} -> acc
+        end
```

Comment:
If we want to be **really defensive** here, we can make it consider crashing checks as successful (so that a poorly written check doesn't cause node restarts)